### PR TITLE
SDIT-923 Lock check PUT /api/bookings/{bookingId}/activities/{activityId}/attendance

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/BookingRepository.java [hunk 1]
```diff
@@ -236,7 +246,7 @@ public Optional<SentenceCalcDates> getBookingSentenceCalcDates(final Long bookin
     /**
      * @param bookingIds Booking IDs
      * @param cutoffDate Omit alerts which have expired before this date-time
-     * @return a list of active alert codes for each booking id
+     * @return a map of active alert codes for each booking id
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/BookingRepository.java:249]: Intellij suggestion

File: src/main/java/uk/gov/justice/hmpps/prison/repository/BookingRepository.java [hunk 2]
```diff
@@ -461,7 +491,7 @@ public Optional<VisitDetails> getBookingVisitNext(final Long bookingId, final Lo
                     BookingRepositorySql.GET_NEXT_BOOKING_VISIT.getSql(),
                     createParams("bookingId", bookingId, "fromDate", DateTimeConverter.fromLocalDateTime(from)),
                     VISIT_ROW_MAPPER);
-            return Optional.of(result)
+            return Optional.ofNullable(result)
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/BookingRepository.java:494]: Intellij suggestion

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 3]
```diff
@@ -298,8 +300,11 @@ public ResponseEntity<Void> updateAttendance(@PathVariable("offenderNo") @Parame
...
-        bookingService.updateAttendance(bookingId, activityId, updateAttendance);
+    public ResponseEntity<Void> updateAttendance(@NotNull @PathVariable("bookingId") @Parameter(description = "The booking Id of the prisoner", required = true, example = "213531") final Long bookingId,
+                                                 @NotNull @PathVariable("activityId") @Parameter(description = "The activity id", required = true, example = "1212131") final Long activityId,
+                                                 @RequestParam(value = "lockTimeout", required = false) @Parameter(description = "Whether to timeout if locked", example = "true") final Boolean lockTimeout,
+                                                 @RequestBody @NotNull @Parameter(required = true) final UpdateAttendance updateAttendance) {
+        bookingService.updateAttendance(bookingId, activityId, updateAttendance, lockTimeout == Boolean.TRUE);
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:307]: very minor but i guess ` lockTimeout == Boolean.TRUE` can just be `lockTimeout `?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java (hunk 3 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:307]: I didnt check this, but i believe it is null when the param is not specified

Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:307]: would be better to default it in the `RequestParam` instead e.g. https://github.com/ministryofjustice/prison-api/blob/669aafe910b68a677a1bd8df1c3979ef1748238d/src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt#L51
```
    @RequestParam(value = "includeAliases", required = false, defaultValue = "false")
```

**Pull request merged**